test/bin/stress_testing.sh

```diff
@@ -0,0 +1,221 @@
+      help
+      exit 1
+    fi
+  elif [ "${ACTION}" == "disable" ]; then
+    if [ "${TOTAL_PARAMS}" -lt 3 ]; then
+      echo -e "ERROR: scenario param is missing"
+      help
+      exit 1
+    elif [ "${TOTAL_PARAMS}" -gt 3 ]; then
+      echo -e "ERROR: value param must be removed"
+      help
+      exit 1
+    fi
+  fi
+}
+
+function tbd {
+  echo -e "This action is not implemented yet"
+}
+
+# bandwidth scenario
+function bandwidth_scenario {
+  bandwidth_scenario_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_bandwidth
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_bandwidth
+  elif [ "${ACTION}" == "get" ]; then
+    get_bandwidth
+  fi
+}
+
+function bandwidth_scenario_ready {
+  # kernel-modules-extra
+  if ( ! dnf list installed kernel-modules-extra &> /dev/null);  then
+    echo -e "ERROR: kernel-modules-extra must be installed:\n> sudo dnf -y install kernel-modules-extra\nthen, reboot your system."
+    exit 1
+  fi
+
+  # wondershaper
+  if ( ! sudo wondershaper -v &> /dev/null);  then
+    wget https://raw.githubusercontent.com/magnific0/wondershaper/master/wondershaper &> /dev/null
+    sudo mv wondershaper /usr/bin/wondershaper &> /dev/null
+    sudo chmod +x /usr/bin/wondershaper &> /dev/null
+    if ( ! sudo wondershaper -v &> /dev/null);  then
+      echo -e "ERROR: Failed to install wondershaper"
+      exit 1
+    fi
+  fi
+
+  # speedtest-cli
+  if ( ! speedtest-cli --help &> /dev/null);  then
+    sudo dnf -y install speedtest-cli &> /dev/null
+    if ( ! speedtest-cli --help &> /dev/null);  then
+      echo -e "ERROR: Failed to install speedtest-cli"
+      exit 1
+    fi
+  fi
+}
+
+function enable_bandwidth {
+  echo -e "Bandwidth scenario enabled: max download and upload rate is ${VALUE} Kbps"
+  sudo wondershaper -a "${INTERFACE}" -d "${VALUE}" -u "${VALUE}"
+}
+
+function disable_bandwidth {
+  echo -e "Bandwidth scenario disabled"
+  sudo wondershaper -a "${INTERFACE}" -c || true
+}
+
+function get_bandwidth {
+  speedtest-cli --simple | tail -n2
+}
+
+# latency scenario
+function latency_scenario {
+  latency_scenario_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_latency
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_latency
+  elif [ "${ACTION}" == "get" ]; then
+    get_latency
+  fi
+}
+
+function latency_scenario_ready {
+  if ( ! dnf list installed kernel-modules-extra &> /dev/null);  then
+    echo -e "ERROR: kernel-modules-extra must be installed:\n> sudo dnf -y install kernel-modules-extra\nthen, reboot your system."
+    exit 1
+  fi
+}
+
+function enable_latency {
+  echo -e "Latency scenario enabled: min latency is ${VALUE}"
+  sudo tc qdisc replace dev "${INTERFACE}" root netem delay "${VALUE}"
+}
+
+function disable_latency {
+  echo -e "Latency scenario disabled"
+  sudo tc qdisc delete dev "${INTERFACE}" root
+}
+
+function get_latency {
+  AVG_LATENCY=$(ping -c 3 8.8.8.8 | tail -1 | awk '{print $4}' | cut -d '/' -f 2 | sed 's/\..*//')
```

Comment:
You could simplify the AVG_LATENCY down to a couple of awk commands and two pipes

ping -c 3 8.8.8.8 | awk '/avg/{print $4}' | awk -F"/" '{printf "%d\n",$2}'

Reply:
thanks, I change the command